### PR TITLE
Resolve #503, Update setup.py and unlock version of dependencies

### LINKED ISSUE #503: Sceptre v2-rc missin internal sub packages during installation.
In Sceptre v2-rc `setup.py` packages are updated manually and does not support newly created/refactored sub-packages.

https://github.com/cloudreach/sceptre/blob/v2-rc/setup.py#L45

For example sub-package `sceptre.config` is not installed now.

### PR DESCRIPTION
**Pull request by 1oglop1**
Resolve #503
* use find_packages instead of manual specification
* Update click to 7.0 < 8.0
* Update packaging to 18.0 < 19.0
* Unlock colorama <1.0.0
* Unlock six <2.0.0

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,12 +12,13 @@
...
 install_requirements = [
     "boto3>=1.3,<2.0",
-    "click==6.7",
+    "click>=7.0<8.0",
     "PyYaml==3.13",
     "Jinja2>=2.8,<3",
-    "packaging==16.8",
-    "colorama==0.3.7",
-    "six==1.11.0"
+    "packaging>=18.0<19.0",
```
Comment (ngfgrant) [inline setup.py]: I am not sure what the consequences of upgrading this package by 2 major version releases will be. Prefer to keep `>=16.0,<17.0` for now.

File: setup.py [hunk 2]
```diff
@@ -12,12 +12,13 @@
 
 install_requirements = [
     "boto3>=1.3,<2.0",
-    "click==6.7",
+    "click>=7.0<8.0",
```
Comment (ngfgrant) [inline setup.py]: I am not sure what the consequences of upgrading this package by 2 major version releases will be. Prefer to keep `>=6.7,<7.0`. Until we can do further test

File: setup.py [hunk 3]
```diff
@@ -12,12 +12,13 @@
...
     "PyYaml==3.13",
     "Jinja2>=2.8,<3",
-    "packaging==16.8",
-    "colorama==0.3.7",
-    "six==1.11.0"
+    "packaging>=18.0<19.0",
+    "colorama>=0.3.7<1.0.0",
```
Comment (ngfgrant) [inline setup.py]: Pre-stable packaging. Would definitely like to get this dependency to a stable release i.e. `1.0`. But for now a lot could have changed between `0.3.7` and `1.0` so I want to keep this pinned to `0.3.7` until we get v2 out and feedback from that has settled.

File: setup.py (hunk 2 above)
Comment (1oglop1) [reply setup.py]: then it will still conflict with awscli and other libs

File: setup.py (hunk 3 above)
Comment (1oglop1) [reply setup.py]: Okay this is fair point. 
So could be updated to latest 0.4.0 ?

Comment (ngfgrant) [reply setup.py]: Again, it is a pre-stable release so there are no guarantees about API compatibility between versions. Indeed looks like they have dropped Python 2.6 support in 0.4.0 which is something we still support. Happy for this to be pinned to `0.3.9` until we drop python 2 support.

File: setup.py (hunk 1 above)
Comment (ngfgrant) [reply setup.py]: `17.0` drops support for python 2.6 which we still support (according to our classifiers)

**Pull request closed without merging**